File: server/BeerWorkshop.Application/Dto/Responses/Checks/CheckResponseDto.cs
using BeerWorkshop.Database.Enums;

namespace BeerWorkshop.Application.Dto.Responses.Checks;

public record CheckResponseDto(
    Guid Id,
    int OrderNumber,
    decimal TotalAmount,
    TransactionType TransactionType,
    DateTime TransactionDate,

    List<CheckItemResponseDto> CheckItems
);

File: client/src/redux/dtos/responses/checks/checkResponseDto.ts
import { TransactionType } from "../../../enums";
import { CheckItemResponseDto } from "./checkItemResponseDto";

export interface CheckResponseDto {
  id: string;
  orderNumber: number;
  totalAmount: number;
  transactionType: TransactionType;
  transactionDate: Date;

  checkItems: CheckItemResponseDto[];
}
